ui/apps/platform/src/Components/SelectSingle/SelectSingle.tsx

```diff
@@ -38,33 +41,75 @@ function SelectSingle({
 }: SelectSingleProps): ReactElement {
     const [isOpen, setIsOpen] = useState(false);
 
-    function onSelect(_event, selection) {
-        // The mouse event is not useful.
-        setIsOpen(false);
-        handleSelect(id, selection);
+    function onSelect(
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        selection: string | number | undefined
+    ) {
+        if (typeof selection === 'string') {
+            setIsOpen(false);
+            handleSelect(id, selection);
+        }
     }
 
+    function onToggle() {
+        setIsOpen(!isOpen);
+    }
+
+    // Find the display text for the selected value
+    const getDisplayText = (): string => {
+        if (!value) {
+            return placeholderText;
+        }
+
+        const selectedChild = children.find((child) => {
+            return child.props.value === value;
+        });
+
+        return (selectedChild?.props.children as string) || value;
+    };
+
+    const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle
+            ref={toggleRef}
+            onClick={onToggle}
+            isExpanded={isOpen}
+            isDisabled={isDisabled}
+            icon={toggleIcon}
+            aria-label={toggleAriaLabel}
+            id={id}
```

Comment:
Does using `id` here and in the `<Select>` component below end up with two elements using the same id value?

Reply:
Also, valid. Ideally, we could have a separate `menuToggleId` prop for the `MenuToggle`'s `id`.